src/Console/Command/CheckCommand.php

```diff
@@ -151,7 +152,12 @@ protected function execute(InputInterface $input, OutputInterface $output)
 
         $output->writeln(sprintf('<comment>There are %s deprecations:</comment>', count($violations)));
 
-        $container['violation.renderer']->renderViolations($violations);
+        $container['violation.renderer.console']->renderViolations($violations);
+
+        if ($htmlOutputFile = $input->getOption('log-html')) {
+            $container['violation.renderer.html']->setOutputFilename($htmlOutputFile);
+            $container['violation.renderer.html']->renderViolations($violations);
```

Comment:
Wouldn't it be cleaner to make the output path an argument to the `renderViolations` method? You're always calling `setOutputFilename` before and force the `HtmlOutputRenderer` to maintain a state that it doesn't actually need.